### PR TITLE
Little ARM case update.

### PR DESCRIPTION
**Pull request by devnexen**
giving hint to cpu it can takes a fast coffee break
to cool down.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -165,6 +165,10 @@ if ("${ARCH}" STREQUAL "x86_64")
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpmath=sse")
 endif()
 
+if ("${ARCH}" STREQUAL "arm")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=native")
+endif()
```
Comment (DanielGibson) [inline CMakeLists.txt]: Are you sure this is a good idea?
Can't this break binaries build for redistribution (e.g. in debian packages)?

**Comment by Yamagi** [comment]
What about the Makefile build?

**Comment by Yamagi** [comment]
Or another question: Is `-march=native` necessary to get the opcode, because gcc defaults to some crappy old ARM revision or is ist to get some more optimizations? In the later case we could leave the CMakeLists.txt and Makefile alone, after we've merged either my *cflags* branch or #523 you do some something like `CFLAGS=-march=native make`.

**Comment by devnexen** [comment]
> Or another question: Is `-march=native` necessary to get the opcode, because gcc defaults to some crappy old ARM revision or is ist to get some more optimizations? In the later case we could leave the CMakeLists.txt and Makefile alone, after we've merged either my _cflags_ branch or #523 you do some something like `CFLAGS=-march=native make`.

Absolutely not optimisation but on ARM mode it can generate this asm instruction otherwise. will update the Makefile

**Comment by Yamagi** [comment]
Hi,
I had a discussion about this with some people that know a lot more than me about ARM. The general consent is: 

* For 32-Bit ARM it should be okay to force `-march=armv6k`. It would cover everything starting from the first RPi generation and older cores are too slow for YQ2 anyway.
* For 64-Bit ARM we could force `-march=armv8.0-a`, it would cover all aarch64 compatible cores.

Forcing the architecture to a sane minimum would simplify the code, because you don't need to communicate into the code if the build supports the `yield` opcode.

**Comment by Yamagi** [comment]
I've just merged a squashed version of these commits into master. Thank you for looking into this, it was a good idea and really good work. :) On my RPi 3 the `yield` opcode saves about 7°C with the dedicated server and 3°C with the gl1 renderer.

**Pull request closed without merging**